rasa/core/featurizers/single_state_featurizer.py

```diff
@@ -40,127 +102,227 @@ class SingleStateFeaturizer:
     featurized into a list of `rasa.utils.features.Features`.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
         """Initialize the single state featurizer."""
-        # rasa core can be trained separately, therefore interpreter during training
-        # will be `RegexInterpreter`. If the model is combined with a rasa nlu model
-        # during prediction the interpreter might be different.
-        # If that is the case, we need to make sure to "reset" the interpreter.
-        self._use_regex_interpreter = False
-        self._default_feature_states = {}
-        self.action_texts = []
-        self.entity_tag_specs = []
-
-    def _create_entity_tag_specs(
-        self, bilou_tagging: bool = False
-    ) -> List[EntityTagSpec]:
-        """Returns the tag to index mapping for entities.
+        self.prepared = False
+        self.use_regex_interpreter = False
+
+        # FIXME: If this is re-instantiated for a policy that we want to fine-tune
+        # then the information to use a regex interpreter will be lost unless
+        # the  NLU Interpreter is still not present during re-training....
+        # Is this a problem?
+
+    def prepare(
+        self, domain: Domain, interpreter: Optional[NaturalLanguageInterpreter]
+    ) -> None:
+        """
+        TODO: all of this looks like it could be done in the init....
+        TODO/FIXME: the item2index_mappings are also needed during inference, which
+          is why it is very confusing this is computed in "prepare_for_training" only...
+        """
+        # TODO: the following this was if...: = True before, is this a problem?
+        self.use_regex_interpreter = isinstance(interpreter, RegexInterpreter)
+        self.item2index_mappings = {
+            key: item2index_mapping(items)
+            for key, items in [
+                (INTENT, domain.intents),
+                (ACTION_NAME, domain.action_names_or_texts),
+                (ENTITIES, domain.entity_states),
+                (SLOTS, domain.slot_states),
+                (ACTIVE_LOOP, domain.form_names),
```

Comment:
I think we should rename `entity_states` to `entity_feature_names` and similar for `slot_states`.

Reply:
yes, agree! Maybe we turn part of that notion page into some keyword index (might also help to trace back State vs Message vs UtterEvent vs...)